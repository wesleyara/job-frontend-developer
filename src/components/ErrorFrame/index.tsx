import { MdErrorOutline } from "react-icons/md";

interface ErrorFrameProps {
  error: Error | null;
}

export const ErrorFrame = ({ error }: ErrorFrameProps) => {
  return (
    <section
      data-cy="error-frame"
      className="mt-5 w-full px-3 flex flex-col items-center gap-3"
    >
      <MdErrorOutline size={50} />
      <span className="text-center">{error?.message}</span>
    </section>
  );
};
