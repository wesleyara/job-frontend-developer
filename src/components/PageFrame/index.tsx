"use client";

import { FaSearch } from "react-icons/fa";

import { Search } from "../Search";

export const PageFrame = () => {
  return (
    <main className="h-[100vh] flex flex-col items-center justify-center bg-[#ebebeb]">
      <Search.Root>
        <Search.Input />
        <Search.Button icon={FaSearch} onClick={() => console.log("teste")} />
      </Search.Root>
    </main>
  );
};
