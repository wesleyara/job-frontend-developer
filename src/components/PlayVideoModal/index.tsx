import { Dialog, DialogPanel, DialogTitle } from "@headlessui/react";
import { YoutubeDetailsType } from "~/types";
import { Dispatch, SetStateAction } from "react";

interface PlayVideoModalProps {
  item: YoutubeDetailsType;
  isOpen: boolean;
  setIsOpen: Dispatch<SetStateAction<boolean>>;
}

export const PlayVideoModal = ({
  item,
  isOpen,
  setIsOpen,
}: PlayVideoModalProps) => {
  return (
    <Dialog
      open={isOpen}
      onClose={() => setIsOpen(false)}
      className="relative z-50"
    >
      <div className="fixed inset-0 flex mx-auto items-center justify-center p-4">
        <DialogPanel className="space-y-4 border border-gray-300 bg-white p-12 max-w-[768px] w-[90%] text-center rounded-md relative">
          <button
            data-cy="close-youtube-modal"
            className="absolute top-2 right-3"
            onClick={() => setIsOpen(false)}
          >
            X
          </button>
          <DialogTitle className="text-lg">{item.snippet.title}</DialogTitle>

          <iframe
            data-cy="youtube-player"
            src={`https://www.youtube.com/embed/${item.id}`}
            title={item.snippet.title}
            className="inset-0 w-full h-[200px] sm:h-[400px]"
            allowFullScreen
          />
        </DialogPanel>
      </div>
    </Dialog>
  );
};
