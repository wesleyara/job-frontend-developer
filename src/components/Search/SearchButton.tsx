"use client";

import { ButtonHTMLAttributes, ElementType } from "react";

interface SearchButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  icon: ElementType;
}

export const SearchButton = ({ icon: Icon, ...rest }: SearchButtonProps) => {
  return (
    <button {...rest} className="p-3 bg-blue-500 text-white rounded-md">
      <Icon />
    </button>
  );
};
