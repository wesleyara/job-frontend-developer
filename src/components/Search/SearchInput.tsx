import { InputHTMLAttributes } from "react";

interface SearchInputProps extends InputHTMLAttributes<HTMLInputElement> {}

export const SearchInput = ({ ...rest }: SearchInputProps) => {
  return (
    <input
      {...rest}
      type="text"
      placeholder="Busque o artista"
      className="py-2 px-3 rounded-md outline-none"
    />
  );
};
