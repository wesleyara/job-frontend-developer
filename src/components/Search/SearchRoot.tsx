import { ReactNode } from "react";

interface SearchRootProps {
  children: ReactNode;
}

export const SearchRoot = ({ children }: SearchRootProps) => {
  return (
    <div className="flex items-center justify-center gap-1">{children}</div>
  );
};
