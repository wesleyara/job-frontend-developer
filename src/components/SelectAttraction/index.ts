import { SelectAttractionButton } from "./SelectAttractionButton";
import { SelectAttractionContent } from "./SelectAttractionContent";
import { SelectAttractionRoot } from "./SelectAttractionRoot";

export const SelectAttraction = {
  Root: SelectAttractionRoot,
  Content: SelectAttractionContent,
  Button: SelectAttractionButton,
};
