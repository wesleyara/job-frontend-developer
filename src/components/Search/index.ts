import { SearchButton } from "./SearchButton";
import { SearchInput } from "./SearchInput";
import { SearchRoot } from "./SearchRoot";

export const Search = {
  Root: SearchRoot,
  Button: SearchButton,
  Input: SearchInput,
};
