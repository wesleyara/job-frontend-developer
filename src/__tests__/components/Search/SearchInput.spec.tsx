import { render } from "@testing-library/react";
import { Search } from "~/components/Search";
import { beforeEach, describe, expect, it } from "vitest";

describe("SearchInput unit test", () => {
  let sut: JSX.Element;

  beforeEach(() => {
    sut = <Search.Input />;
  });

  it("should be defined", () => {
    expect(sut).toBeDefined();
  });

  it("should render correctly", () => {
    const { container } = render(sut);

    expect(container).toMatchSnapshot();
  });

  it("should have an input element", () => {
    const { container } = render(sut);

    expect(container.querySelector("input")).not.toBeNull();
  });

  it("should have a placeholder", () => {
    const { container } = render(sut);

    expect(container.querySelector("input")?.getAttribute("placeholder")).toBe(
      "Busque o artista",
    );
  });

  it("should have a type of text", () => {
    const { container } = render(sut);

    expect(container.querySelector("input")?.getAttribute("type")).toBe("text");
  });

  it("should type in the input", () => {
    const { container } = render(sut);
    const input = container.querySelector("input");

    expect(input).not.toBeNull();

    input?.setAttribute("value", "teste");

    expect(input?.getAttribute("value")).toBe("teste");
  });

  it("should have a tailwind class name", () => {
    const { container } = render(sut);

    const element = container.querySelector(".rounded-md");

    expect(element).not.toBeNull();
  });
});
