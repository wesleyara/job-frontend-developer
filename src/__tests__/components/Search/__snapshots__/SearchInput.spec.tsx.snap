// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchInput unit test > should render correctly 1`] = `
<div>
  <input
    class="py-2 px-3 rounded-md outline-none"
    placeholder="Busque o artista"
    type="text"
  />
</div>
`;
